### ISSUE TITLE
All default pseudo class rules replaced by custom one

### ISSUE THREAD

**Issue by VincenzoSMC** [chakra-ui/chakra-ui#5583]
### Description

All default pseudo class rules replaced by custom one

### Link to Reproduction

https://codesandbox.io/s/beautiful-hoover-td66s?file=/src/index.tsx

### Steps to reproduce

2. Click on '...'
3. Scroll down to '...'

### Chakra UI Version

1.6.4

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

When you try to add a custom pseudo class rule, that one replaces the whole default pseudo class rules instead of adding it.

ex.:

<Tab
_selected: {
  color: "black".   // this rule is the only one rendered
}
></Tab>

**Comment by TimKolberger** [comment]
This is currently the intended behavior to keep the styles readable.
The style props are just a shorthand way to override some CSS definitions.
As @takethefake pointed out: By deep merging the style we would not only introduce a O(n^2) algorithm (very slow) but also adding another cascading layer of CSS which would make it even harder to find the correct place to alter the styles.

A solution for your issue is to define a separate chakra component by using the c[hakra factory](https://chakra-ui.com/docs/features/chakra-factory) function to define a new component.

**Comment by takethefake** [comment]
@TimKolberger what do you think if we do `Object.assign` for pseudos? this way we keep complexity low and the user doesn't need to redefine the whole pseudo when overwriting?

**Comment by TimKolberger** [comment]
I assume you meant `mergeWith` for pseudo props?

I discussed this with @segunadebayo  the other day and he pointed out that this is intentional. And since then I share the opinion to override the nested style object, because otherwise it is hard to follow the style definitions (which ones are applied, which ones are getting overridden).

There is no browser CSS debugger for merging JS objects, so we would need to merge all the styles in our heads to keep track of what is happening. And I am not able to do that, even on a good day 😄

IMO the way to go is to structure the styles as clean as possible. By adding some limitations chakra can help enforcing this for for us. And explicitly not to merge pseudo props is one of those limitations.

**Comment by takethefake** [comment]
We're closing this for now to avoid unmanageable complexity

**Comment by bertho-zero** [comment]
I'm facing the same problem, I understand the fact that you have to keep the system simple but defining a component or a variant is not always relevant when you just want to complete a particular case with an additional property.

To use the example cited, I need to create a component just to add one property to `_selected` for a special case:
```tsx
function MyTab({
  size,
  variant,
  colorScheme,
  _selected,
  children,
  ...chakraProps
}) {
  const styles = useMultiStyleConfig("Tabs", { size, variant, colorScheme });

  return (
    <Tab
      _selected={{
        ...styles.tab._selected,
        ..._selected
      }}
      {...chakraProps}
    >
      {children}
    </Tab>
  );
}

// in App component:
<MyTab
  _selected={{
    color: "black" // same here
  }}
>
  tab1
</MyTab>
```

What do you think of having a syntax that allows properties to be added rather than replaced, leaving the choice depending on the case?

```tsx
<Tab
  __selected={{ // __ = assign; _ = replace (or vice versa)
    color: "black"
  }}
>
  tab1
</Tab>
```
